In [1]:

#carga de librerias
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns 
import datetime as dt



In [2]:
# se lee el archivo a analizar
df_2017purchaseprices = pd.read_csv("DataSet/2017PurchasePricesDec.csv")


In [3]:
#se muestran las primeras 5 filas para comprobar la carga de datos
df_2017purchaseprices.head()


Brand                  Description  Price   Size Volume  Classification  \
0     58  Gekkeikan Black & Gold Sake  12.99  750mL    750               1   
1     62     Herradura Silver Tequila  36.99  750mL    750               1   
2     63   Herradura Reposado Tequila  38.99  750mL    750               1   
3     72         No. 3 London Dry Gin  34.99  750mL    750               1   
4     75    Three Olives Tomato Vodka  14.99  750mL    750               1   

   PurchasePrice  VendorNumber                   VendorName  
0           9.28          8320  SHAW ROSS INT L IMP LTD      
1          28.67          1128  BROWN-FORMAN CORP            
2          30.46          1128  BROWN-FORMAN CORP            
3          26.11          9165  ULTRA BEVERAGE COMPANY LLP   
4          10.94          7245  PROXIMO SPIRITS INC.

In [4]:
df_2017purchaseprices.shape

(12261, 9)

In [5]:
#Normalizacion de datos de las columnas Size y Volume

# Obtener los valores únicos de la columna 'Size'
unique_sizes = df_2017purchaseprices['Size'].unique()

# Obtener los valores únicos de la columna 'Volume'
unique_volumes = df_2017purchaseprices['Volume'].unique()

# Mostrar los valores únicos de 'Size'
print("Valores únicos de 'Size':")
print(unique_sizes)

# Mostrar los valores únicos de 'Volume'
print("Valores únicos de 'Volume':")
print(unique_volumes)


Valores únicos de 'Size':
['750mL' '1000mL' '1750mL' '50mL' '375mL' '100mL 4 Pk' '50mL 5 Pk' '100mL'
 '200mL' '50mL 4 Pk' '50mL 3 Pk' '300mL' '200mL 4 Pk' 'Unknown'
 '750mL 2 Pk' '250mL 4 Pk' '1500mL' '3000mL' '5000mL' '4000mL'
 '187mL 4 Pk' '150mL' '187mL' '500mL' '720mL' '650mL' '200mL 3 Pk'
 '187mL 3 Pk' '330mL' '250mL' '750mL + 3/' '18000mL' '180mL' '750mL + 4/'
 '6000mL' '1000mL 2 Pk' '750mL 12 P' '750mL 6 Pk' '20mL 5 Pk' nan
 '375mL 2 Pk' '20000mL' '50mL 12 Pk' '750mL 3 Pk' '375mL 3 Pk'
 '750mL + 1/' '200mL 5 Pk' '162.5mL' '400mL' '1100ml' '600mL' '19500mL'
 '560mL' '3750mL' '750mL  3' '1500ml' '9000mL']
Valores únicos de 'Volume':
['750' '1000' '1750' '50' '375' '100' '200' '300' 'Unknown' '250' '1500'
 '3000' '5000' '4000' '187' '150' '500' '720' '650' '330' '18000' '180'
 '6000' '20' nan '20000' '162.5' '400' '1100' '600' '19500' '560' '3750'
 '9000']


In [6]:
# Filtrar filas donde 'Size' contiene "Pk"
pk_sizes = df_2017purchaseprices[df_2017purchaseprices['Size'].str.contains('Pk', na=False)]

# Obtener un ejemplo único de cada valor de 'Size' que contiene "Pk"
unique_pk_sizes = pk_sizes.drop_duplicates(subset=['Size'])

# Mostrar los ejemplos únicos
print(unique_pk_sizes[['Size', 'Description', 'Classification']])


              Size                   Description  Classification
113     100mL 4 Pk  Glenmorangie Collection 4 Pk               1
115      50mL 5 Pk      Malibu 50ml Sampler Pack               1
330      50mL 4 Pk  Jack Daniels Family 4 Pk/50m               1
363      50mL 3 Pk              Crown Royal Trio               1
398     200mL 4 Pk   Johnnie Walker Collection 4               1
641     750mL 2 Pk  Bombay Sapphire & East 2 Pak               1
932     250mL 4 Pk    Goslings Dark'N Stormy RTD               1
1480    187mL 4 Pk     Sutter Home Clsc Wh Znfdl               2
3690    200mL 3 Pk  St George Gin 3/200mls Combo               1
3762    187mL 3 Pk        Cavit Prosecco Lunetta               2
6918   1000mL 2 Pk  Grey Goose 2Pak Cherry/Origi               1
7737    750mL 6 Pk    Viberti Mixed Wooden Box 6               2
7858     20mL 5 Pk  The Bitter Truth 5/20mls Pak               1
8754    375mL 2 Pk       Maker's Mark Combo Pack               1
8864    50mL 12 Pk    12 

In [7]:
# Filtrar las filas donde 'Size' es 'Unknown' y obtener los valores de 'Brand'
brands_with_unknown_size = df_2017purchaseprices[df_2017purchaseprices['Size'] == 'Unknown']['Brand']

# Mostrar los resultados
print(brands_with_unknown_size)

542       2993
5921      9908
8795      8992
10009    90590
Name: Brand, dtype: int64


In [8]:
#Rellenar los datos Unknown de la columna Size
# Se actualiza el valor del Brand 2993, y al resto se les agrega no aplica ya que se trata de tazas y vasos y de un producto descatalogado.
# Actualizar la columna 'Size' para Brand 2993 a 200
df_2017purchaseprices.loc[df_2017purchaseprices['Brand'] == 2993, 'Size'] = '200'

# Actualizar la columna 'Size' para Brand 9908, 8992 y 90590 a 'No aplica'
df_2017purchaseprices.loc[df_2017purchaseprices['Brand'].isin([9908, 8992, 90590]), 'Size'] = 'No Aplica'

# Mostrar las filas actualizadas para verificar el cambio
updated_rows = df_2017purchaseprices[df_2017purchaseprices['Brand'].isin([2993, 9908, 8992, 90590])]
print(updated_rows)


       Brand                   Description  Price       Size   Volume  \
542     2993             Angostura Bitters   7.49        200  Unknown   
5921    9908      Tito's Copper Mug 2 Pack  21.01  No Aplica  Unknown   
8795    8992                      Group 92   1.99  No Aplica  Unknown   
10009  90590  Overture Champagne 2Glass Pk  19.95  No Aplica  Unknown   

       Classification  PurchasePrice  VendorNumber  \
542                 1           5.39          5895   
5921                1          16.15          4425   
8795                1           1.43          1703   
10009               2          13.12          4425   

                                VendorName  
542            Mizkan Americas, Inc.        
5921                 MARTIGNETTI COMPANIES  
8795   ALISA CARR BEVERAGES                 
10009                MARTIGNETTI COMPANIES  


In [9]:
# Se procede a separa la columna de Size en dos columnas (una d eSize y otra de Packs)
# Definir una función para procesar la columna 'Size'
def process_size(size):
    if pd.isna(size) or 'Unknown' in size or 'No Aplica' in size:
        return pd.Series(['No Aplica', 'No Aplica'])
    
    size_parts = size.split(' ')
    size_ml = size_parts[0].strip()
    pack = ''
    
    if len(size_parts) > 1:
        for pack_part in size_parts[1:]:
            pack += pack_part
    else:
        pack = 'No Aplica'
    
    return pd.Series([size_ml, pack])

# Aplicar la función de conversión a la columna 'Size'
df_2017purchaseprices[['Size', 'Pack']] = df_2017purchaseprices['Size'].apply(process_size)

# Mostrar las primeras filas del DataFrame para verificar el cambio
print(df_2017purchaseprices[['Size', 'Pack']].head(400))


      Size       Pack
0    750mL  No Aplica
1    750mL  No Aplica
2    750mL  No Aplica
3    750mL  No Aplica
4    750mL  No Aplica
..     ...        ...
395  750mL  No Aplica
396  750mL  No Aplica
397  750mL  No Aplica
398  200mL        4Pk
399  750mL  No Aplica

[400 rows x 2 columns]


In [10]:
# Filtrar registros donde 'Pack' es diferente de 'No Aplica'
filtered_df = df_2017purchaseprices[df_2017purchaseprices['Pack'] != 'No Aplica']

# Mostrar los registros filtrados
filtered_df.head(50)


Brand                   Description  Price   Size Volume  \
113    1133  Glenmorangie Collection 4 Pk  31.99  100mL    100   
115    1136      Malibu 50ml Sampler Pack   4.99   50mL     50   
160    1365   Stoli Mini Bar 5/50mLs Pack   7.99   50mL     50   
330    2250  Jack Daniels Family 4 Pk/50m  12.99   50mL     50   
363    2453              Crown Royal Trio   5.99   50mL     50   
398    2604   Johnnie Walker Collection 4  99.99  200mL    200   
641    3341  Bombay Sapphire & East 2 Pak  34.99  750mL    750   
932    4062    Goslings Dark'N Stormy RTD   7.99  250mL    250   
940    4105  Patron XO Cafe Variety 3 Pak   6.99   50mL     50   
1127   4881    Bacardi Twin Pack 2/750mls  19.99  750mL    750   
1315   5714    Jose Cuervo Lime Margarita   9.99  200mL    200   
1422   6477  Avion Tequila 3/50ml Sampler   9.99   50mL     50   
1480   6861     Sutter Home Clsc Wh Znfdl   5.99  187mL    187   
1484   6876        Sutter Home Clsc Chard   5.99  187mL    187   
1717   8513  Baileys Irish Cream Mini's 4  10.99  100mL    100   
2488  20446    Barefoot Bubbly Brut Cuvee   9.99  187mL    187   
2649  22800  Alexander Vly Wicked Weekend  15.99  750mL    750   
2851  29128        Sutter Home Merlot Cal   5.99  187mL    187   
3144  39599      R M Woodbridge Chard Cal   7.99  187mL    187   
3690   1259  St George Gin 3/200mls Combo  27.99  200mL    200   
3710   4535   Bacardi Limon & Mango 2 Pak  19.99  750mL    750   
3762  12203        Cavit Prosecco Lunetta   9.99  187mL    187   
3779  14670           Sutter Home Moscato   5.99  187mL    187   
3847  25522     Sutter Home Clsc Cab Svgn   5.99  187mL    187   
3872  33334   Cavit Pinot Grigio Delle Ve   8.99  187mL    187   
3892  39598   R M Woodbridge Cab Svgn Cal   7.99  187mL    187   
3893  39600     R M Woodbridge Merlot Cal   7.99  187mL    187   
3894  39652    Sutter Home Pnt Grigio Cal   5.99  187mL    187   
3897  40175              M & R Asti 4 Pak  11.99  187mL    187   
3907  41974     Beringer Classic Wh Znfdl   6.99  187mL    187   
3909  42256               Duboeuf Bjls Rd   9.49  187mL    187   
3974    159     Glenmorangie Original VAP  34.99  750mL    750   
4200  20447  Barefoot Bubbly Pink Moscato   9.99  187mL    187   
4341  39798     R M Woodbridge Pnt Grigio   7.99  187mL    187   
4537  40118     F Coppola Sophia Bl de Bl  13.99  187mL    187   
4551   2502  Dewars Wh Label & Honey 2 Pk  34.99  750mL    750   
4704   9116       KAH Tequila 3/50mLs Pak  24.99   50mL     50   
5052  13696       Barefoot Cellars Pnt Nr   6.49  187mL    187   
5102  18853  Barefoot Cellars Pink Moscat   6.49  187mL    187   
5304   1520   Pinnacle Vodka Cane 4/50mls   3.99   50mL     50   
5356  46165       Barefoot Pnt Grigio Cal   6.49  187mL    187   
5432    285       Red Stag VAP 750+4/50ml  18.99  750mL    750   
5530  13702     Barefoot Cellars Cab Svgn   6.49  187mL    187   
5589  13703      Barefoot Cellars Moscato   6.49  187mL    187   
5899   6456      Sutter Home Clsc Svgn Bl   5.99  187mL    187   
6092  17040      Sutter Home Pink Moscato   5.99  187mL    187   
6106  17486  Barefoot Cellars Red Moscato   6.49  187mL    187   
6530  23205      M & R Prosecco Frizzante  15.99  187mL    187   
6597  23665                    Cooks Brut   7.99  187mL    187   
6598  23666     Cooks Extra Dry Champagne   7.99  187mL    187   

      Classification  PurchasePrice  VendorNumber  \
113                1          23.01          8112   
115                1           3.87         17035   
160                1           6.05          8664   
330                1           9.69          1128   
363                1           4.31          3960   
398                1          73.52          3960   
641                1          25.17           480   
932                1           5.83          1485   
940                1           5.07          9165   
1127               1          14.81           480   
1315               1           7.14          7245   
142

In [11]:

# Encontrar filas con 'Unknown' en la columna 'Volume'
unknown_rows = df_2017purchaseprices[df_2017purchaseprices['Volume'] == 'Unknown']

# Encontrar filas con NaN en la columna 'Volume'
nan_rows = df_2017purchaseprices[df_2017purchaseprices['Volume'].isna()]

# Mostrar filas con 'Unknown' en la columna 'Volume'
print("Filas con 'Unknown' en la columna 'Volume':")
print(unknown_rows)

# Mostrar filas con NaN en la columna 'Volume'
print("\nFilas con NaN en la columna 'Volume':")
print(nan_rows)


Filas con 'Unknown' en la columna 'Volume':
       Brand                   Description  Price       Size   Volume  \
542     2993             Angostura Bitters   7.49        200  Unknown   
5921    9908      Tito's Copper Mug 2 Pack  21.01  No Aplica  Unknown   
8795    8992                      Group 92   1.99  No Aplica  Unknown   
10009  90590  Overture Champagne 2Glass Pk  19.95  No Aplica  Unknown   

       Classification  PurchasePrice  VendorNumber  \
542                 1           5.39          5895   
5921                1          16.15          4425   
8795                1           1.43          1703   
10009               2          13.12          4425   

                                VendorName       Pack  
542            Mizkan Americas, Inc.        No Aplica  
5921                 MARTIGNETTI COMPANIES  No Aplica  
8795   ALISA CARR BEVERAGES                 No Aplica  
10009                MARTIGNETTI COMPANIES  No Aplica  

Filas con NaN en la columna 'Volume':


Este producto se analiza mas abajo 
Brand Description  Price Size Volume  Classification  PurchasePrice  \
7915   4202         NaN    0.0  NaN    NaN               1          11.19   

Brand                   Description  Price     Size   Volume  \
542     2993             Angostura Bitters   7.49  Unknown  Unknown   
5921    9908      Tito's Copper Mug 2 Pack  21.01  Unknown  Unknown   
8795    8992                      Group 92   1.99  Unknown  Unknown   
10009  90590  Overture Champagne 2Glass Pk  19.95  Unknown  Unknown  

De estos 4 productos con Volumen y tamaño desconocido es porque dos no son liquidos( vasos y tazas) y a los otros dos se procede a agregarle la informacion faltante(uno esta descatalogado y al otro s ele agrega el volumen correspondiente)
  

In [12]:

# Actualizar la columna 'Volume' para Brand 2993 a 200
df_2017purchaseprices.loc[df_2017purchaseprices['Brand'] == 2993, 'Volume'] = '200'

# Actualizar la columna 'Volume' para Brand 9908, 8992 y 90590 a 'No aplica'
df_2017purchaseprices.loc[df_2017purchaseprices['Brand'].isin([9908, 8992, 90590]), 'Volume'] = 'No aplica'

# Mostrar las filas actualizadas para verificar el cambio
updated_rows = df_2017purchaseprices[df_2017purchaseprices['Brand'].isin([2993, 9908, 8992, 90590])]
print(updated_rows)


       Brand                   Description  Price       Size     Volume  \
542     2993             Angostura Bitters   7.49        200        200   
5921    9908      Tito's Copper Mug 2 Pack  21.01  No Aplica  No aplica   
8795    8992                      Group 92   1.99  No Aplica  No aplica   
10009  90590  Overture Champagne 2Glass Pk  19.95  No Aplica  No aplica   

       Classification  PurchasePrice  VendorNumber  \
542                 1           5.39          5895   
5921                1          16.15          4425   
8795                1           1.43          1703   
10009               2          13.12          4425   

                                VendorName       Pack  
542            Mizkan Americas, Inc.        No Aplica  
5921                 MARTIGNETTI COMPANIES  No Aplica  
8795   ALISA CARR BEVERAGES                 No Aplica  
10009                MARTIGNETTI COMPANIES  No Aplica  


In [13]:
# Verificar si hay filas duplicadas
duplicated_rows = df_2017purchaseprices.duplicated()

# Contar el número de filas duplicadas
num_duplicated_rows = duplicated_rows.sum()

# Mostrar el número de filas duplicadas
print(f'Número de filas duplicadas en df_2017purchaseprices: {num_duplicated_rows}')

# Opcional: Mostrar las filas duplicadas
if num_duplicated_rows > 0:
    print("Filas duplicadas en df_2017purchaseprices:")
    print(df_2017purchaseprices[duplicated_rows])

Número de filas duplicadas en df_2017purchaseprices: 0


In [14]:
# se verifica el tipo de dato de cada columna
print(df_2017purchaseprices.dtypes)

Brand               int64
Description        object
Price             float64
Size               object
Volume             object
Classification      int64
PurchasePrice     float64
VendorNumber        int64
VendorName         object
Pack               object
dtype: object


In [15]:
#Verificamos la cantidad de datos faltantes por columna

print(df_2017purchaseprices.isnull().sum())

Brand             0
Description       1
Price             0
Size              0
Volume            1
Classification    0
PurchasePrice     0
VendorNumber      0
VendorName        0
Pack              0
dtype: int64


In [16]:
# Filtrar la fila donde la columna 'Description' es nula
null_description_row = df_2017purchaseprices[df_2017purchaseprices['Description'].isnull()]

# Mostrar la fila con 'Description' nulo
print(null_description_row)


#se concluye que las celdas con valores faltantes corresponden al mismo producto

      Brand Description  Price       Size Volume  Classification  \
7915   4202         NaN    0.0  No Aplica    NaN               1   

      PurchasePrice  VendorNumber                   VendorName       Pack  
7915          11.19           480  BACARDI USA INC              No Aplica  


In [17]:
# Se analiza si la fila con valores faltantes se han olvidado de eliminarla por alguna razon 
#  Filtrar las filas donde 'VendorNumber' es igual a 480 y 'PurchasePrice' es igual a 11.19
filtered_rows = df_2017purchaseprices[(df_2017purchaseprices['VendorNumber'] == 480) & 
                                      (df_2017purchaseprices['PurchasePrice'] == 11.19)]

# Mostrar el resultado
print(filtered_rows)


       Brand             Description  Price       Size Volume  Classification  \
7915    4202                     NaN   0.00  No Aplica    NaN               1   
10418   4092  Bacardi Grapefruit Rum  13.99      750mL    750               1   

       PurchasePrice  VendorNumber                   VendorName       Pack  
7915           11.19           480  BACARDI USA INC              No Aplica  
10418          11.19           480  BACARDI USA INC              No Aplica  


Nos encontramos con que ya existe un producto de ese proveedor con el mismo precio por lo que se procede a eliminar el Brand 4202, se entiende que se trata del mismo producto

In [18]:
# Eliminar las filas donde la columna 'Description' tiene valores nulos
df_2017purchaseprices.dropna(subset=['Description'], inplace=True)

# Verificar que se haya eliminado la fila con valor nulo

print(df_2017purchaseprices.isnull().sum())

Brand             0
Description       0
Price             0
Size              0
Volume            0
Classification    0
PurchasePrice     0
VendorNumber      0
VendorName        0
Pack              0
dtype: int64


In [19]:
# Verificar si hay valores negativos en las columnas numéricas
negatives = df_2017purchaseprices.select_dtypes(include=['number']) < 0

# Obtener los nombres de las columnas que tienen valores negativos
negative_columns = negatives.any()

# Filtrar las columnas que tienen al menos un valor negativo
columns_with_negatives = negative_columns[negative_columns].index.tolist()

# Mostrar los resultados
if columns_with_negatives:
    print(f"Columnas con valores negativos: {columns_with_negatives}")
    for column in columns_with_negatives:
        print(f"\nValores negativos en la columna '{column}':")
        print(df_2017purchaseprices[df_2017purchaseprices[column] < 0])
else:
    print("No hay valores negativos en el DataFrame.")

No hay valores negativos en el DataFrame.


In [20]:
#Se hace un analsis de las distintas columnas 

for i in df_2017purchaseprices.columns:
    print(i)
    print(df_2017purchaseprices[i].describe()) #descripcion basica 
    print('--------------------------------------')

Brand
count    12260.000000
mean     17990.191680
std      12528.395592
min         58.000000
25%       5991.500000
50%      18789.000000
75%      25117.250000
max      90631.000000
Name: Brand, dtype: float64
--------------------------------------
Description
count                    12260
unique                   11114
top       Jagermeister Liqueur
freq                         7
Name: Description, dtype: object
--------------------------------------
Price
count    12260.000000
mean        38.643392
std        206.159284
min          0.000000
25%         10.990000
50%         15.990000
75%         29.990000
max      13999.900000
Name: Price, dtype: float64
--------------------------------------
Size
count     12260
unique       35
top       750mL
freq       9840
Name: Size, dtype: object
--------------------------------------
Volume
count     12260
unique       33
top         750
freq       9840
Name: Volume, dtype: object
--------------------------------------
Classification
count  

**Description**

count                    12260
unique                   11114
top       Jagermeister Liqueur
freq                         7

**VendorName**
count                     12260
unique                      136
top       MARTIGNETTI COMPANIES
freq                       1631

In [21]:
# Se analizan las columnas con tipo de datos flotantes
# Lista de columnas de tipo flotante
columns_of_interest = ['Brand', 'Price','Classification', 'PurchasePrice', 'VendorNumber']

# Función para imprimir las estadísticas de una columna
def print_column_stats(df, column):
    print(f"\nEstadísticas para la columna '{column}':")
    print(f"Número total de valores (incluyendo nulos): {len(df[column])}")
    column_stats = df[column].describe()
    print(f"Número total de valores: {column_stats['count']}")
    print(f"Número de valores únicos: {df[column].nunique()}")
    print(f"Valor más frecuente (top): {df[column].mode()[0]}")
    print(f"Frecuencia del valor más frecuente (freq): {df[column].value_counts().iloc[0]}")
    print(f"Valor mínimo: {column_stats['min']}")
    print(f"Valor máximo: {column_stats['max']}")
    print(f"Media: {column_stats['mean']}")


# Iterar sobre las columnas de interés y mostrar las estadísticas
for column in columns_of_interest:
    print_column_stats(df_2017purchaseprices, column)



Estadísticas para la columna 'Brand':
Número total de valores (incluyendo nulos): 12260
Número total de valores: 12260.0
Número de valores únicos: 12260
Valor más frecuente (top): 58
Frecuencia del valor más frecuente (freq): 1
Valor mínimo: 58.0
Valor máximo: 90631.0
Media: 17990.19168026101

Estadísticas para la columna 'Price':
Número total de valores (incluyendo nulos): 12260
Número total de valores: 12260.0
Número de valores únicos: 380
Valor más frecuente (top): 9.99
Frecuencia del valor más frecuente (freq): 703
Valor mínimo: 0.0
Valor máximo: 13999.9
Media: 38.643391517128876

Estadísticas para la columna 'Classification':
Número total de valores (incluyendo nulos): 12260
Número total de valores: 12260.0
Número de valores únicos: 2
Valor más frecuente (top): 2
Frecuencia del valor más frecuente (freq): 8693
Valor mínimo: 1.0
Valor máximo: 2.0
Media: 1.7090538336052201

Estadísticas para la columna 'PurchasePrice':
Número total de valores (incluyendo nulos): 12260
Número total 

Conclusiones Basadas en las Estadísticas de las Columnas

***Brand:***
Número total de valores: 12,260
Número de valores únicos: 12,260
Cada valor es único, lo cual sugiere que no hay marcas repetidas. Esto podría indicar que la columna contiene identificadores únicos para cada marca en lugar de nombres de marca. 

***Price:***
Valor más frecuente (top): 9.99
Frecuencia del valor más frecuente (freq): 703
Valor mínimo: 0.0
Valor máximo: 13,999.9
Media: 38.64
La amplia gama de precios, desde 0 hasta casi 14,000, sugiere una gran variabilidad en los precios de los productos. 

***Classification:***
Número de valores únicos: 2
Valor más frecuente (top): 2
Valor mínimo: 1.0
Valor máximo: 2.0
La mayoría de los productos están clasificados como 2, lo que indica que esta clasificación es la más común. Hay solo dos valores únicos en esta columna, sugiriendo una clasificación binaria o muy limitada de los productos.

***PurchasePrice:***
Número de valores únicos: 2,314
Valor más frecuente (top): 6.53
Frecuencia del valor más frecuente (freq): 62
Valor mínimo: 0.0
Valor máximo: 11,111.03
Media: 26.49


***VendorNumber:***
Número de valores únicos: 131
Valor más frecuente (top): 4425
Frecuencia del valor más frecuente (freq): 1,639
Hay 131 proveedores únicos, con el proveedor más común (4425) apareciendo 1,639 veces. 


In [22]:
# Se observan que hay celdas con Purchase Price y Price = 0
# Contar el número de celdas con precio igual a 0 en la columna 'PurchasePrice'
zero_price_count = (df_2017purchaseprices['PurchasePrice'] == 0).sum()

# Mostrar el resultado
print(f"Número de celdas con precio igual a 0 en 'PurchasePrice': {zero_price_count}")


Número de celdas con precio igual a 0 en 'PurchasePrice': 1


In [23]:
# Filtrar las filas donde 'PurchasePrice' y 'Price' es igual a 0
zero_price_rows = df_2017purchaseprices[df_2017purchaseprices['PurchasePrice'] == 0]

# Mostrar las filas filtradas
print(zero_price_rows)



       Brand                  Description  Price   Size Volume  \
10803   2166  The Macallan Double Cask 12    0.0  750mL    750   

       Classification  PurchasePrice  VendorNumber  \
10803               1            0.0          2561   

                        VendorName       Pack  
10803  EDRINGTON AMERICAS           No Aplica  


In [24]:
#Se concluye que se trata del mismo producto que tiene Price y PurchasePrice =0

In [25]:

# Se le pide al cliente el PurchasePrice y el Price de "The Macallan Double Cask 12" para agregarlo en el data set
# Verificar las filas que contienen "The Macallan Double Cask 12" en la columna Description
macallan_rows = df_2017purchaseprices[df_2017purchaseprices['Description'].str.contains("The Macallan Double Cask 12", na=False)]
print("\nFilas antes de la actualización:")
print(macallan_rows)

# Actualizar el valor de PurchasePrice a 50 y Price a 75 para "The Macallan Double Cask 12"
df_2017purchaseprices.loc[df_2017purchaseprices['Description'].str.contains("The Macallan Double Cask 12", na=False), ['PurchasePrice', 'Price']] = [50, 75]

# Verificar las filas que contienen "The Macallan Double Cask 12" después de la actualización
macallan_rows_updated = df_2017purchaseprices[df_2017purchaseprices['Description'].str.contains("The Macallan Double Cask 12", na=False)]
print("\nFilas después de la actualización:")
print(macallan_rows_updated)





Filas antes de la actualización:
       Brand                  Description  Price   Size Volume  \
10803   2166  The Macallan Double Cask 12    0.0  750mL    750   

       Classification  PurchasePrice  VendorNumber  \
10803               1            0.0          2561   

                        VendorName       Pack  
10803  EDRINGTON AMERICAS           No Aplica  

Filas después de la actualización:
       Brand                  Description  Price   Size Volume  \
10803   2166  The Macallan Double Cask 12   75.0  750mL    750   

       Classification  PurchasePrice  VendorNumber  \
10803               1           50.0          2561   

                        VendorName       Pack  
10803  EDRINGTON AMERICAS           No Aplica  


In [26]:
# Analisis sobre el significado de las bebidas clasificadas como 1 y las clasificadas como 2
# Obtener estadísticas descriptivas agrupadas por clasificación
grouped_by_class = df_2017purchaseprices.groupby('Classification').describe(include='all')

# Mostrar estadísticas descriptivas agrupadas por clasificación
print("Estadísticas Descriptivas Agrupadas por Clasificación:")
print(grouped_by_class)

# Obtener ejemplos representativos de cada clasificación
examples_class_1 = df_2017purchaseprices[df_2017purchaseprices['Classification'] == 1].head(10)
examples_class_2 = df_2017purchaseprices[df_2017purchaseprices['Classification'] == 2].head(10)

print("\nEjemplos Representativos de Clasificación 1:")
print(examples_class_1)

print("\nEjemplos Representativos de Clasificación 2:")
print(examples_class_2)


Estadísticas Descriptivas Agrupadas por Clasificación:
                 Brand                                                   \
                 count unique top freq          mean          std   min   
Classification                                                            
1               3567.0    NaN NaN  NaN   3641.196243  2352.774355  58.0   
2               8693.0    NaN NaN  NaN  23878.017140  9996.909932  82.0   

                                           ...   Pack                        \
                    25%      50%      75%  ... unique        top  freq mean   
Classification                             ...                                
1                1983.0   3345.0   5057.0  ...     11  No Aplica  3523  NaN   
2               17623.0  22765.0  26867.0  ...      5  No Aplica  8629  NaN   

                                              
                std  min  25%  50%  75%  max  
Classification                                
1               NaN  NaN  NaN  N

Diferencias Clave
Clasificación 1 parece estar dominada por bebidas espirituosas (vodkas, tequilas, gins), generalmente vendidas por proveedores grandes y establecidos en tamaños estándar de 750ml y 1000ml.
Clasificación 2 incluye principalmente vinos y algunos licores especiales, con una mayor diversidad de proveedores y tamaños de botellas, lo que sugiere una categoría más diversa en términos de productos y precios.

In [27]:
df_2017purchaseprices.head()

Brand                  Description  Price   Size Volume  Classification  \
0     58  Gekkeikan Black & Gold Sake  12.99  750mL    750               1   
1     62     Herradura Silver Tequila  36.99  750mL    750               1   
2     63   Herradura Reposado Tequila  38.99  750mL    750               1   
3     72         No. 3 London Dry Gin  34.99  750mL    750               1   
4     75    Three Olives Tomato Vodka  14.99  750mL    750               1   

   PurchasePrice  VendorNumber                   VendorName       Pack  
0           9.28          8320  SHAW ROSS INT L IMP LTD      No Aplica  
1          28.67          1128  BROWN-FORMAN CORP            No Aplica  
2          30.46          1128  BROWN-FORMAN CORP            No Aplica  
3          26.11          9165  ULTRA BEVERAGE COMPANY LLP   No Aplica  
4          10.94          7245  PROXIMO SPIRITS INC.         No Aplica

In [28]:
# Cambio de nombre a las columnas
# Diccionario que mapea los nombres actuales de las columnas a los nuevos nombres
column_renames = {
    'Brand': 'ProductoID',
    'Description': 'Descripcion',
    'Price': 'Precio_venta',
    'Size': 'Tamaño',
    'Volume': 'Volumen',
    'Classification': 'Clasificacion',
    'PurchasePrice': 'Precio_compra',
    'VendorNumber': 'ProveedorID',
    'VendorName': 'Proveedor',
    'Pack': 'Pack'
}

# Renombrar las columnas
df_2017purchaseprices.rename(columns=column_renames, inplace=True)

# Mostrar las primeras filas del DataFrame para verificar los cambios
print(df_2017purchaseprices.head())

   ProductoID                  Descripcion  Precio_venta Tamaño Volumen  \
0          58  Gekkeikan Black & Gold Sake         12.99  750mL     750   
1          62     Herradura Silver Tequila         36.99  750mL     750   
2          63   Herradura Reposado Tequila         38.99  750mL     750   
3          72         No. 3 London Dry Gin         34.99  750mL     750   
4          75    Three Olives Tomato Vodka         14.99  750mL     750   

   Clasificacion  Precio_compra  ProveedorID                    Proveedor  \
0              1           9.28         8320  SHAW ROSS INT L IMP LTD       
1              1          28.67         1128  BROWN-FORMAN CORP             
2              1          30.46         1128  BROWN-FORMAN CORP             
3              1          26.11         9165  ULTRA BEVERAGE COMPANY LLP    
4              1          10.94         7245  PROXIMO SPIRITS INC.          

        Pack  
0  No Aplica  
1  No Aplica  
2  No Aplica  
3  No Aplica  
4  No Aplic

In [29]:
#Crear un nuevo archivo de Producto_proveedor
# Seleccionar las columnas deseadas
columnas_seleccionadas = df_2017purchaseprices[['ProductoID', 'ProveedorID', 'Precio_compra', 'Precio_venta']].copy()

# Agregar la columna ProductoProveedorId con valores autonumerados que empiezan en 1
columnas_seleccionadas['Producto_proveedorID'] = range(1, len(columnas_seleccionadas) + 1)

# Reordenar las columnas para que ProductoProveedorId sea la primera columna
columnas_seleccionadas = columnas_seleccionadas[['Producto_proveedorID', 'ProductoID', 'ProveedorID', 'Precio_compra', 'Precio_venta']]

# Guardar el nuevo dataframe en un archivo CSV
ruta_nuevo_archivo = 'ruta_al_archivo/nuevo_archivo.csv'
columnas_seleccionadas.to_csv('DataSetFiltrados/PRODUCTO_PROVEEDOR.csv', index=False)


In [30]:
# Agrupar por ProveedorID y contar los Proveedor únicos
proveedor_counts = df_2017purchaseprices.groupby('ProveedorID')['Proveedor'].nunique()

# Filtrar los ProveedorID que tienen más de un Proveedor
proveedores_multiples = proveedor_counts[proveedor_counts > 1]

# Mostrar los ProveedorID que tienen más de un Proveedor
print(proveedores_multiples.index.tolist())

[1587, 1703, 2000, 4425]


In [31]:
# Elimina espacios en blanco al inicio y al final de los nombres de los proveedores
df_2017purchaseprices['Proveedor'] = df_2017purchaseprices['Proveedor'].str.strip()

# # Reemplaza el nombre del proveedor en la columna 'Proveedor'
df_2017purchaseprices['Proveedor'] = df_2017purchaseprices['Proveedor'].replace('VINEYARD BRANDS LLC', 'VINEYARD BRANDS INC')

# Verifica que el reemplazo se haya realizado correctamente
print(df_2017purchaseprices[df_2017purchaseprices['Proveedor'] == 'VINEYARD BRANDS INC'].head())

# Reemplaza el nombre del proveedor en la columna 'Proveedor' donde ProveedorID es 2000
df_2017purchaseprices.loc[(df_2017purchaseprices['ProveedorID'] == 2000) & 
                       (df_2017purchaseprices['Proveedor'] == 'SOUTHERN GLAZERS W&S OF NE'), 
                       'Proveedor'] = 'SOUTHERN WINE & SPIRITS NE'

# Verifica que el reemplazo se haya realizado correctamente
print("Después del reemplazo:")
print(df_2017purchaseprices[df_2017purchaseprices['ProveedorID'] == 2000].head())

# Seleccionar las columnas 'ProveedorID' y 'Proveedor'
df_proveedores = df_2017purchaseprices[['ProveedorID', 'Proveedor']]

# Eliminar filas duplicadas para asegurarse de que cada proveedor sea único
df_proveedores = df_proveedores.drop_duplicates()

# Agregar el nuevo registro utilizando loc
df_proveedores.loc[len(df_proveedores)] = [201359, 'FLAVOR ESSENCE INC']

# Guardar el resultado en un nuevo archivo CSV llamado 'Proveedores.csv'
df_proveedores.to_csv('DataSetFiltrados/PROVEEDOR.csv', index=False)

# Mostrar el DataFrame resultante para verificar
print(df_proveedores.head())


      ProductoID                   Descripcion  Precio_venta Tamaño Volumen  \
1520        7302   Marques de Caceres Rioja Re         10.99  750mL     750   
1569        8016     La Vieille Ferme Wh Rhone          6.99  750mL     750   
1570        8018  La Vieille Ferme Red Ventoux          6.99  750mL     750   
1987       13497        Mohua Svgn Bl Marlboro         10.99  750mL     750   
1988       13498          Riva Rey Tempranillo          6.99  750mL     750   

      Clasificacion  Precio_compra  ProveedorID            Proveedor  \
1520              2           6.42         1587  VINEYARD BRANDS INC   
1569              2           4.79         1587  VINEYARD BRANDS INC   
1570              2           4.60         1587  VINEYARD BRANDS INC   
1987              2           6.80         1587  VINEYARD BRANDS INC   
1988              2           4.63         1587  VINEYARD BRANDS INC   

           Pack  
1520  No Aplica  
1569  No Aplica  
1570  No Aplica  
1987  No Aplica  
19

In [32]:
#Se crea la tabla final de PRODUCTO

# Seleccionar las columnas deseadas
df_2017purchaseprices_final = df_2017purchaseprices[['ProductoID', 'Descripcion', 'Tamaño', 'Volumen', 'Clasificacion', 'Pack']]



In [33]:
# Guardar el resultado en un archivo CSV para revisarlo
df_2017purchaseprices_final.to_csv('DataSetFiltrados/PRODUCTO.csv', index=False)

In [34]:
df_2017purchaseprices_final.head()

ProductoID                  Descripcion Tamaño Volumen  Clasificacion  \
0          58  Gekkeikan Black & Gold Sake  750mL     750              1   
1          62     Herradura Silver Tequila  750mL     750              1   
2          63   Herradura Reposado Tequila  750mL     750              1   
3          72         No. 3 London Dry Gin  750mL     750              1   
4          75    Three Olives Tomato Vodka  750mL     750              1   

        Pack  
0  No Aplica  
1  No Aplica  
2  No Aplica  
3  No Aplica  
4  No Aplica